src/arch/arm/pmu.hh

```diff
@@ -358,28 +359,40 @@ class PMU : public SimObject, public ArmISA::BaseISADevice
         }
 
       protected:
-        struct RegularProbe: public  ProbeListenerArgBase<uint64_t>
+        struct RegularProbe: public ProbeListenerArgBase<uint64_t>
         {
-            RegularProbe(RegularEvent *parent, SimObject* obj,
-                std::string name)
-                : ProbeListenerArgBase(obj->getProbeManager(), name),
-                  parentEvent(parent) {}
+            RegularProbe(RegularEvent *parent, std::string name)
+                : ProbeListenerArgBase(std::move(name)), parentEvent(parent) {}
 
             RegularProbe() = delete;
 
-            void notify(const uint64_t &val);
+            void notify(const uint64_t &val) override;
 
           protected:
             RegularEvent *parentEvent;
         };
+        struct CacheProbe: public ProbeListenerArgBase<CacheAccessProbeArg>
+        {
+            CacheProbe(RegularEvent *parent, std::string name)
+                : ProbeListenerArgBase(std::move(name)), parentEvent(parent) {}
+            CacheProbe() = delete;
+
+            void notify(const CacheAccessProbeArg &val) override {
```

Comment:
Nit: this shoul be:

```
void
notify(const CacheAccessProbeArg &val) override
{
```

Reply:
Ah, sorry I contribute to too many different projects to always remember the right coding style... I wish we could get #362 landed. Will fix the style issues.